src/main/java/design/MazeDesignerInteractor.java

```diff
@@ -0,0 +1,47 @@
+package design;
+
+import entities.DesignableMaze;
+
+public class MazeDesignerInteractor implements MazeDesignerInputBoundary {
+
+    DesignableMaze dm;
+
+    public void newMaze(){
+        dm = new DesignableMaze(11, 17);
+    }
+
+    public void resetMaze(){
+        dm.emptySetup();
+    }
+
+    public void buildWall(int row, int col){
+        dm.placeWall(row, col);
+    }
+    public void removeWall(int row, int col){
+        if (row != 0 && col != 0 && row != dm.getNumRow()-1 && col != dm.getNumCol()-1) {
+            dm.deleteWall(row, col);
+        }
+    }
+    public void startPoint(int row, int col){
+        dm.placeStart(row, col);
+    }
+    public void endPoint(int row, int col){
+        dm.placeEnd(row, col);
+    }
+    public void randomMaze(){
+        RandomizedPrim mg = new RandomizedPrim();
+        mg.generate(dm);
```

Comment:
The generate function should not be called with the maze instance, since when the RandomizedPrim instance is initialized, it already stores the designable maze:

`mg.generate()`